packages/fetchai/connections/oef/connection.py

```diff
@@ -534,16 +271,17 @@ def on_search_result(self, search_id: int, agents: List[Address]) -> None:
         :param agents: the list of agents.
         :return: None
         """
-        assert self.in_queue is not None
-        assert self.loop is not None
+        self._check_loop_and_queue()
         oef_search_dialogue = self.oef_msg_id_to_dialogue.pop(search_id, None)
         if oef_search_dialogue is None:
-            logger.warning("Could not find dialogue for search_id={}".format(search_id))
-            return
+            logger.warning(
+                "Could not find dialogue for search_id={}".format(search_id)
+            )  # pragma: nocover
+            return  # pragma: nocover
         last_msg = oef_search_dialogue.last_incoming_message
         if last_msg is None:
-            logger.warning("Could not find last message.")
-            return
+            logger.warning("Could not find last message.")  # pragma: nocover
```

Comment:
if you place `pragma: no cover` after the if it works better

Reply:
yes, but the main idea do not hide all if block from coverage, in case i'd like to add some other code into this if, I can forget to remove  nocover directive and new code will be skipped from coverage as well, so decided to use nocover only for specific line